probe/kubernetes/reporter.go

```diff
@@ -474,16 +421,16 @@ func (r *Reporter) podTopology(services []Service, replicaSets []ReplicaSet, dae
 			report.MakeServiceNodeID(service.UID()),
 		))
 	}
-	for _, replicaSet := range replicaSets {
-		selector, err := replicaSet.Selector()
+	for _, deployment := range deployments {
+		selector, err := deployment.Selector()
 		if err != nil {
 			return pods, err
 		}
 		selectors = append(selectors, match(
-			replicaSet.Namespace(),
+			deployment.Namespace(),
 			selector,
-			report.ReplicaSet,
-			report.MakeReplicaSetNodeID(replicaSet.UID()),
+			report.Deployment,
+			report.MakeDeploymentNodeID(deployment.UID()),
```

Comment:
How does this work?

Reply:
`match` will add the deployment as the pod's parent if the following conditions are true:
* the pod and the deployment belong to the same namespace
* the pod and the deployment match the same labels.

Reply:
https://github.com/weaveworks/scope/blob/master/probe/kubernetes/reporter.go#L403

Reply:
Deployments, ReplicaSets and Pods will be assigned the same label in the manifest.
The label is required: https://github.com/kubernetes/kubernetes/issues/28475